app/Models/Story.php

```diff
@@ -151,4 +151,68 @@ public function registerMediaCollections(): void
 
         $this->addMediaCollection('images')->acceptsMimeTypes(config('services.media-library.allowed_image_formats'));
     }
-}
\ No newline at end of file
+
+    /**
+     * Add a response to the story customer_request field
+     * @return void
+     */
+    public function addCustomerResponse($response)
+    {
+        $sender = auth()->user();
+
+        $senderType = '';
+
+        if ($this->status == StoryStatus::Done) {
+            throw new \Exception('Cannot add response to a done story');
+        }
+        $style = '';
+
+        if ($sender->id == $this->user_id) {
+            $senderType = 'developer';
+            $style = "style='background-color: #f8f9fa; border-left: 4px solid #6c757d; padding: 10px 20px;'";
+        } else if ($sender->id == $this->tester_id) {
+            $senderType = 'tester';
+            $style = "style='background-color: #e6f7ff; border-left: 4px solid #1890ff; padding: 10px 20px;'";
+        } else if ($sender->id == $this->creator_id) {
+            $senderType = 'customer';
+            $style = "style='background-color: #fff7e6; border-left: 4px solid #ffa940; padding: 10px 20px;'";
+        } else {
+            throw new \Exception('User is not allowed to add a response to this story');
+        }
+
+        $formattedResponse = $sender->name . " ha risposto il: " . now()->format('d-m-Y H:i') . "\n <div $style> <p>" . $response . " </p> </div> <div style='height: 2px; background-color: #e2e8f0; margin: 20px 0;'></div>";
+        $this->customer_request = $formattedResponse . $this->customer_request;
+        $this->save();
+
+        switch ($senderType) {
+            case 'developer':
+                if ($this->tester_id) {
+                    if ($this->tester_id != $this->user_id) {
+                        Mail::to($this->tester->email)->send(new \App\Mail\StoryResponse($this, $this->tester, $sender, $response));
+                    }
+                }
+                if ($this->creator_id) {
```

Comment:
estrarre questo codice dallo switch e metterlo prima modificando la guardia nel seguente modo if ($this->creator_id && $senderType != 'customer') {